### PR TITLE
feat: enhance mitre tag regex in sigma test

### PR DESCRIPTION
**Pull request by nasbench**
Tightening the MITRE tag check regex a little bit.

cc @phantinuss

### CODE REVIEW

**Comment by frack113** [comment]
https://attack.mitre.org/docs/ATTACK_Design_and_Philosophy_March_2020.pdf give  : 
attack.t####
attack.t####.###
attack.g####
attack.s####

File: tests/test_rules.py [hunk 1]
```diff
@@ -79,7 +79,7 @@ def test_legal_trademark_violations(self):
     def test_optional_tags(self):
         files_with_incorrect_tags = []
         tags_pattern = re.compile(
-            r"cve\.\d+\.\d+|attack\.t\d+\.*\d*|attack\.[a-z_]+|car\.\d{4}-\d{2}-\d{3}")
+            r"cve\.\d+\.\d+|(attack\.t\d{4}\.\d{3}$|attack\.t\d{4}$)|attack\.[a-z_]+|car\.\d{4}-\d{2}-\d{3}")
```
Comment (frack113) [inline tests/test_rules.py:82]: attack\.t\d{4}\.\d{3}$|attack\.[gts]\d{4}$

Comment (phantinuss) [reply tests/test_rules.py:82]: All of it is fine in my opinion. We can also add the groups and ?software?

Comment (nasbench) [reply tests/test_rules.py:82]: I will add the `g` and `s` yes

Comment (nasbench) [inline tests/test_rules.py:82]:
```suggestion
            r"cve\.\d+\.\d+|(attack\.t\d{4}\.\d{3}$|attack\.[gts]\d{4}$)|attack\.[a-z_]+|car\.\d{4}-\d{2}-\d{3}")
```

File: tests/test_rules.py [hunk 2]
```diff
@@ -79,7 +79,7 @@ def test_legal_trademark_violations(self):
     def test_optional_tags(self):
         files_with_incorrect_tags = []
         tags_pattern = re.compile(
-            r"cve\.\d+\.\d+|attack\.t\d+\.*\d*|attack\.[a-z_]+|car\.\d{4}-\d{2}-\d{3}")
+            r"cve\.\d+\.\d+|(attack\.t\d{4}\.\d{3}$|attack\.[gts]\d{4}$)|attack\.[a-z_]+|car\.\d{4}-\d{2}-\d{3}")
```
Comment (phantinuss) [inline tests/test_rules.py:82]:
```suggestion
            r"cve\.\d+\.\d+|attack\.(t\d{4}\.\d{3}|[gts]\d{4})$|attack\.[a-z_]+|car\.\d{4}-\d{2}-\d{3}")
```

Comment (phantinuss) [reply tests/test_rules.py:82]: It's the same xD

Comment (nasbench) [reply tests/test_rules.py:82]: We removed one attack so we gain some bits :D

**Review by frack113 — APPROVED** [review]

**Pull request merged**